### DISCUSSION TITLE
function signature is missing in the documentation

### DISCUSSION THREAD

**Discussion by 15r10nk** [mkdocstrings/mkdocstrings#495] [Q&A] [answered]
I am not able to get the function signature shown in my documentation

function:
``` python
def foo(a:int):
    """
    foo is missing its signature in the docs
    """
    print(a)
```

documentation:
``` markdown
::: mymodule.foo
```

the documentation looks like follow

![grafik](https://user-images.githubusercontent.com/44680962/203868798-011d488c-776d-4927-9df1-e150f7d7a433.png)

I would expect an `foo(a:int)` at the top of the documentation.

The repo to reproduce this issue is [here](https://github.com/15r10nk/bug_mkdocstrings)

Use `poetry run mkdocs serve` to reproduce the issue.

I still think that it is most likely some stupid mistake on my side, but I don't find it.

**Comment by pawamoy** [comment, marked as answer]
(converted to discussion)

Hi! Thanks so much for the repro! It's so pleasant to investigate something when there's a repository I can clone, and even more when it provides the lock file for dependencies! Thanks thanks thanks

So, indeed, you're lacking a particular option here: `show_root_heading`.
That's because I made the assumption that, in general, users would document `::: module` rather than `::: module.function`.
So by default the "root heading" (the heading for the top-level object specified in `:::`) is not shown, in this case the function signature.

Try this:

```
::: mymodule.foo
    options:
      show_root_heading: yes
```

Or globally, in mkdocs.yml:

```yml
plugins:
- mkdocstrings:
    handlers:
      python:
        options:
          show_root_heading: yes
```
